Controller/Environments/BlackwoodForest/Cards/DesolationCardController.cs

```diff
@@ -37,70 +34,31 @@ public override void AddTriggers()
 
         public override IEnumerator Play()
         {
-            IEnumerator choiceRoutine = base.DoActionToEachTurnTakerInTurnOrder(
-                ttc => ttc.IsHero && !ttc.IsIncapacitatedOrOutOfGame,
-                ChoiceResponse);
+            int damageToDeal = Game.H + 1;
 
-            if (base.UseUnityCoroutines)
-            {
-                yield return base.GameController.StartCoroutine(choiceRoutine);
-            }
-            else
+            IEnumerable<Function> Functions(HeroTurnTakerController httc) => new Function[2]
             {
-                base.GameController.ExhaustCoroutine(choiceRoutine);
-            }
-        }
+                new Function(httc, "Discard all but 1 card", SelectionType.DiscardCard,
+                    () => base.GameController.SelectCardsFromLocationAndMoveThem(httc, httc.HeroTurnTaker.Hand,
+                        httc.NumberOfCardsInHand - 1, httc.NumberOfCardsInHand - 1, 
+                        new LinqCardCriteria(card => card.Location == httc.HeroTurnTaker.Hand, "hand"),
+                        new[] { new MoveCardDestination(httc.TurnTaker.Trash) }, cardSource: base.GetCardSource())),
 
-        private IEnumerator ChoiceResponse(TurnTakerController ttc)
-        {
-            List<YesNoCardDecision> yesNoResults = new List<YesNoCardDecision>();
-            IEnumerator yesNoRoutine = base.GameController.MakeYesNoCardDecision(ttc.ToHero(),
-                SelectionType.DiscardCard, base.Card, storedResults: yesNoResults, cardSource: base.GetCardSource());
+                new Function(httc, $"Take {damageToDeal} psychic damage", SelectionType.DealDamage,
+                    () => this.DealDamage(this.Card, httc.CharacterCard, damageToDeal, DamageType.Psychic, cardSource: base.GetCardSource()))
```

Comment:
```suggestion
                    () => this.DealDamage(this.Card, (Card c) => httc.CharacterCards.Contains(c) && c.IsTarget, damageToDeal, DamageType.Psychic, cardSource: base.GetCardSource()))
```

Keep this bit from breaking on the Sentinels